File: homework/thinhbka/lesson5-graph/841_Keys_and_Rooms.py
# https://leetcode.com/problems/keys-and-rooms/
class Solution(object):
    def canVisitAllRooms(self, rooms):
        seen = [False] * len(rooms)
        seen[0] = True
        stack = [0]
        cnt = 0
        while stack:
            node = stack.pop() 
            cnt+=1
            for nei in rooms[node]:
                if not seen[nei]: 
                    seen[nei] = True
                    stack.append(nei) 
        return cnt == len(rooms)

File: homework/Truong_Duong/Lesson_7/841_Keys_and_Rooms.js
// Solution for https://leetcode.com/problems/keys-and-rooms/

/**
 * @param {number[][]} rooms
 * @return {boolean}
 */
 var canVisitAllRooms = function(rooms) {
    const N = rooms.length;
    const visited = new Array(N).fill(false);

    const dfs = function(s) {
        visited[s] = true;
        count += 1;
        
        for(const v of rooms[s]) {
            if(!visited[v]) dfs(v);
        }
    }
    
    var count = 0;
    dfs(0);
    
    return count === rooms.length;
};